src/main/java/org/jenkinsci/plugins/p4/tasks/AbstractTask.java

```diff
@@ -28,23 +28,33 @@
 	private static Logger logger = Logger.getLogger(AbstractTask.class.getName());
 
 	private final P4BaseCredentials credential;
+	private final String credentialName;
 	private final TaskListener listener;
 
 	private Workspace workspace;
 
 	@Deprecated
-	public AbstractTask(String credential, TaskListener listener) {
-		this.credential = ConnectionHelper.findCredential(credential);
+	public AbstractTask(String credentialName, TaskListener listener) {
+		this.credential = ConnectionHelper.findCredential(credentialName);
+		this.credentialName = credentialName;
 		this.listener = listener;
 	}
 
-	public AbstractTask(String credential, Item project, TaskListener listener) {
-		this.credential = ConnectionHelper.findCredential(credential, project);
+	public AbstractTask(P4BaseCredentials p4Credential, String credentialName, TaskListener listener) {
```

Comment:
Abstract is a core file to the plugin - everything goes through here.  We are going to have to be ultra careful with any changes. I'm not keen having both the class and name - just choose one.